### PR TITLE
KAFKA-2413: fix ConsumerCoordinator updateConsumer

### PR DESCRIPTION
**Pull request by onurkaraman**

### CODE REVIEW

**Comment by SinghAsDev** [comment]
+1, LGTM! However, you might have to fix the test failure.

**Comment by onurkaraman** [comment]
Thanks for the review! I'll additionally tweak the assertion message to compare the actual and expected results.

It also looks like testHWCheckpointWithFailuresMultipleLogSegments java.util.NoSuchElementException: None.get is a known transitive issue: [KAFKA-2415](https://issues.apache.org/jira/browse/KAFKA-2415)

**Comment by ijuma** [comment]
A couple of style comments, LGTM otherwise.

**Comment by eribeiro** [comment]
@onurkaraman Don't forget to assign the issue to yourself at JIRA!

**Comment by hachikuji** [comment]
@onurkaraman Would it be worthwhile to add a test case for shrinking the subscription set?

**Comment by onurkaraman** [comment]
Hey Jason. Yeah I was thinking the same! I'll add it now.

File: core/src/test/scala/integration/kafka/api/ConsumerTest.scala [hunk 1]
```diff
@@ -254,6 +254,24 @@ class ConsumerTest extends IntegrationTestHarness with Logging {
...
+      this.consumers(0).poll(50)
+      this.consumers(0).subscriptions == subscriptions.asJava
+    }, s"Expected partitions ${subscriptions.asJava} but actually got ${this.consumers(0).subscriptions}")
+
+    TestUtils.createTopic(this.zkClient, otherTopic, 2, serverCount, this.servers)
+    this.consumers(0).subscribe(otherTopic)
+    TestUtils.waitUntilTrue(() => {
+      this.consumers(0).poll(50)
+      this.consumers(0).subscriptions == expandedSubscriptions.asJava
+    }, s"Expected partitions ${subscriptions.asJava} but actually got ${this.consumers(0).subscriptions}")
```
Comment (onurkaraman) [inline core/src/test/scala/integration/kafka/api/ConsumerTest.scala]: I messed up here: ${subscriptions.asJava} should be ${expandedSubscriptions.asJava} . Too much copy pasta.

**Comment by hachikuji** [comment]
LGTM

**Comment by onurkaraman** [comment]
@ijuma : should I squash the commits and force push? I'm not a fan of having these multiple commits, but I read that this would remove all the earlier comments from the pull request.

It looks like recent git commit messages in the git log show all the intermediate commits, which is something I'd like to avoid.

**Comment by ijuma** [comment]
@onurkaraman, good question. My preference is the same as yours, to squash any commits that don't add any value. However, this is not a requirement of the Kafka project at this stage, so it's up to the submitter.

There are two options if you want to squash:
- Force push to the same branch. Comments on commits may disappear (sometimes they stay with a note about being for an out of date commit), but comments on the PR itself will remain. I personally think this is OK for PRs like this where the comments were all addressed anyway.
- Create a new PR and reference the previous one. This is better for complex PRs where there are a lot of comments.

**Comment by onurkaraman** [comment]
@ijuma thanks. The squash and force push worked!

**Pull request closed without merging**